hivemind/dht/node.py

```diff
@@ -368,7 +368,7 @@ def _update_cache_on_store(self, key_id: DHTID, subkeys: List[Subkey], binary_va
             self.protocol.cache.store(key_id, stored_value_bytes, stored_expiration)
         elif not store_succeeded and not is_dictionary:  # store rejected, check if local cache is also obsolete
             rejected_value, rejected_expiration = max(zip(binary_values, expirations), key=lambda p: p[1])
-            if (self.protocol.cache.get(key_id)[1] or float("inf")) <= rejected_expiration:  # cache would be rejected
+            if (self.protocol.cache.get(key_id) or (None, float("inf")))[1] <= rejected_expiration:  # cache would be rejected
```

Comment:
Great catch!

I am a bit confused as for what could cause this error to occur.

AFAIU, the key is written into the cache in .store_many before this method is called, and the error occurs the value expires *before* the store is finished -- or is removed due to cache size limit.

Is this what triggered the error in your case?

Reply:
Thanks!

The error was triggered when a node has refused to store a record that has not passed the validator (in this case, `store_succeeded = False` as well).

Reply:
Personally, I think that this way of obtaining the expiration time and intentionally failing the check otherwise (with `"inf"`) is kinda hacky and error-prone, which allowed the bug to appear in the first place. 

What do you think about rewriting this in a more clear and explicit manner, such as 
```
cached_value = self.protocol.cache.get(key_id)
if cached_value is not None and cached_value.expiration_time <= rejected_expiration: # cache would be rejected
```

Reply:
Sure!

Personally, I'd argue for prohibiting the `value or default` expressions in all new code since this is known to be a error-prone anti-pattern.